controllers/comments.js

```diff
@@ -127,29 +120,105 @@ export default {
             createdAt: comment.createdAt,
             updatedAt: comment.updatedAt,
             body: comment.body,
-            highlitedText: comment.highlitedText,
+            highlightedText: comment.hightedText,
             author,
             likes: getLikes.count
           };
         })
       );
-      // console.log(structuredComments);
+      // console.log(formatedComments);
       return res.status(200).json({
         status: res.statusCode,
-        comments: structuredComments,
-        commentsCount: structuredComments.length
+        comments: formatedComments,
+        commentsCount: formatedComments.length
       });
     } catch (e) {
-      if (e.message) {
-        return res.status(500).json({
+      return res.status(500).json({
+        status: res.statusCode,
+        error: e.message || '',
+        message: 'Something happened on the server'
+      });
+    }
+  },
+
+  /**
+   * An author of a comment is allowed to update a comment
+   * When a comment is updated we keep the  old comment edits
+   * so that a user can be able to see edit history
+   * @param {object} req
+   * @param {object} res
+   * @return {object} res
+   *
+   */
+  update: async (req, res) => {
+    const { body } = req.body;
+    const { slug, commentId } = req.params;
+    const { id } = req.user;
+
+    if (!body || body.trim() === '') {
+      return res.status(400).json({
+        status: res.statusCode,
+        error: 'Comment body is required'
+      });
+    }
+
+    try {
+      const CommentedArticle = await article.findOne({
+        where: { slug },
+        attributes: ['id'],
+        include: {
+          model: User,
+          attributes: ['username', 'id']
+        }
+      });
+
+      if (!CommentedArticle) {
+        return res.status(400).json({
           status: res.statusCode,
-          message: 'Something happened on the server',
-          error: e.message
+          error: 'Article not found'
         });
       }
+
+      const updatedComment = await Comments.update(
+        { body },
+        {
+          where: {
+            id: commentId,
+            post: CommentedArticle.id,
+            author: id
+          },
+          returning: true
+        }
+      );
+
+      if (updatedComment[0] === 0) {
+        return res.status(400).json({
+          status: res.statusCode,
+          error: 'Comment to update not found'
+        });
+      }
+
+      const editHistory = await CommentEdits.findAll({
+        where: { commentId },
+        attributes: ['body', 'createdAt']
+      });
+      const comment = updatedComment[1][0];
```

Comment:
Also consider the comment on the line `154`

Reply:
The comment updated here doesn't have highlighted text.